src/main/java/soot/Scene.java

```diff
@@ -1176,11 +1176,11 @@ public SootClass getSootClassUnsafe(String className, boolean phantomNonExist) {
     RefType type = nameToClass.get(className);
     if (type != null) {
       synchronized (type) {
-        if (type.hasSootClass() || !SootClass.INVOKEDYNAMIC_DUMMY_CLASS_NAME.equals(className)) {
```

Comment:
Why did you remove this check?

Reply:
There was no other behavior guarded by the check for invokedynamic class, which is why I removed it altogether (but it's still there in the next if statement as before)
![image](https://user-images.githubusercontent.com/35702512/171871617-0c9a2f51-afda-4f66-9e56-eebdb4414af2.png)